src/ducks/groups/helpers.js

```diff
@@ -22,3 +22,40 @@ export const buildVirtualGroups = accounts => {
 
   return virtualGroups
 }
+
+/**
+ * Translate group properties
+ * @param {Object} group - The group to translate
+ * @param {Function} translate - The translation function
+ * @returns {Object} The translated group
+ */
+export const translateGroup = (group, translate) => {
+  return {
+    ...group,
+    label: group.virtual
+      ? translate(`Data.accountTypes.${group.label}`)
+      : group.label
+  }
+}
+/**
+ * Sort groups on their translated label. But always put "others accounts" last
+ * @param {Object[]} groups - The groups to sort
+ * @param {Function} translate - The translation function
+ * @returns {Object[]} The sorted groups
+ */
+export const sortGroups = (groups, translate) => {
+  const [othersGroup] = groups.filter(g => g.virtual && g.label === 'undefined')
+
+  const sortedGroups = sortBy(
+    groups
+      .filter(g => g !== othersGroup)
+      .map(g => translateGroup(g, translate)),
+    g => g.label
+  )
```

Comment:
The sort function can be improved by checking if the group is special (ie `label === undefined`) and returning `Infinity` if it does. This way there is no need for the find and the push at the end.

```
g => g.label === undefined ? Infinity : g.label 
```